tests/utilities.py

```diff
@@ -226,6 +227,93 @@ def get_user_next_id(remote_id):
     return next_id
 
 
+def wait_for_resource_removal_in_db(
+    table, index, identifier, max_attempts=10, delay=0.5
+):
+    """Polls rethinkdb for the requested resource until it has been removed.
+    Useful when commiting a delete transaction in sawtooth and waiting for the
+    resource to be removed from rethink for dependent chained transactions.
+    Args:
+        table:
+            str:    the name of a table to query for the resource in.
+        index:
+            str:    the index of the identifier to query for.
+        identifier:
+            str:    A next_id for a given resource to wait for.
```

Comment:
if that ID field is interchangeable between every object we use then we might want to change the name in the objects themselves to make that clear rather than introduce new names here?